### ISSUE TITLE
Build error when opt-out static optimatization and providing custom Error page

### ISSUE THREAD

**Issue by piotrrussw** [vercel/next.js#22907]
### What version of Next.js are you using?

10.0.8

### What version of Node.js are you using?

14.x.x

### What browser are you using?

-

### What operating system are you using?

linux

### How are you deploying your application?

other

### Describe the Bug

After upgrading to next.js 10.0.8 I can't build my application. I get  the following error:

```
Error occurred prerendering page "/404". Read more: https://err.sh/next.js/prerender-error
TypeError: res.writeHead is not a function
```

I have opt-out static optimatization in `_app.js` and provide custom `_error.js` page.
The following error throws during build process and affects pages: `/404`, `/404.html`, `/500`
I think it might be connected with [#22139](https://github.com/vercel/next.js/pull/22139)

### Expected Behavior

To be able to build page

### To Reproduce

You can try same configuration as in [sandbox](https://codesandbox.io/s/practical-elbakyan-s3xyx)

Not sure if it is a bug or it requires docs update.

Best,
Piotr

**Comment by lipoolock** [comment]
More or less same issue on my side

I extracted content of `ctx` below, diplayed just before the error occurs
```
[  ==] info  - Generating static pages (0/7)headers {
  err: undefined,
  req: {
    url: '/500',
    headers: {},
    getHeader: [Function: getHeader],
    setHeader: [Function: setHeader],
    hasHeader: [Function: hasHeader],
    removeHeader: [Function: removeHeader],
    getHeaderNames: [Function: getHeaderNames]
  },
  res: {
    headers: {},
    getHeader: [Function: getHeader],
    setHeader: [Function: setHeader],
    hasHeader: [Function: hasHeader],
    removeHeader: [Function: removeHeader],
    getHeaderNames: [Function: getHeaderNames]
  },
  pathname: '/_error',
  query: {},
  asPath: '/500',
  AppTree: [Function: AppTree]
}

Error occurred prerendering page "/de/500". Read more: https://err.sh/next.js/prerender-error
```

I do not expect having any URL like `/de/500` on my application, so I don't understand why NextJs is trying to pre-render this...

**Comment by ijjk** [comment]
Hi, this looks like a duplicate of https://github.com/vercel/next.js/issues/22815 which is fixed in `v10.0.9-canary.3`, please update and give it a try!

**Comment by lipoolock** [comment]
Hi @ijjk 
Thanks a lot, you are right, that solves the issue and that was a duplicate for me :+1:
